spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceInitializer.java

```diff
@@ -193,13 +196,28 @@ private void runScripts(List<Resource> resources, String username, String passwo
 		for (Resource resource : resources) {
 			populator.addScript(resource);
 		}
+		DataSource dataSource = resolveDataSource(username, password);
+		DatabasePopulatorUtils.execute(populator, dataSource);
+	}
+
+	private DataSource resolveDataSource(String username, String password) {
 		DataSource dataSource = this.dataSource;
-		if (StringUtils.hasText(username) && StringUtils.hasText(password)) {
-			dataSource = DataSourceBuilder.create(this.properties.getClassLoader())
-					.driverClassName(this.properties.determineDriverClassName()).url(this.properties.determineUrl())
-					.username(username).password(password).build();
+		if (StringUtils.hasText(username)) {
+			try {
+				String url = JdbcUtils.extractDatabaseMetaData(this.dataSource, "getURL");
```

Comment:
It feels to me that this is the wrong approach to the problem. Looking at the reported issue, setting `spring.datasource.schema-username` should not have any impact on an embedded database that we've created. 

Rather, we should detect such case and ignore the username and password.

Reply:
I can add a validation that if `isEmbedded()` then `*-username` and `*-password` are ignored. Indeed, looking at the [comment](https://github.com/spring-projects/spring-boot/pull/19327#issuecomment-562846662) I left, that was one the approaches. WDYT?

Reply:
Unfortunately, the fact the database is embedded is not exactly the same as us replacing the database with an embedded database. 

In theory, I'd very surprised if an embedded database would require different credentials for schema initialization so that approach would work but not really semantically correct IMO.